packages/sdk/src/evm/core/classes/marketplacev3-direct-listings.ts

```diff
@@ -172,15 +172,33 @@ export class MarketplaceV3DirectListings<TContract extends DirectListingsLogic>
     let batches = await getAllInBatches(
       start,
       end,
-      this.contractWrapper.readContract.getAllValidListings,
+      this.contractWrapper.readContract.getAllListings, // TODO: revert to previous setup
```

Comment:
don't think we should do this... would reduce performance quite a bit for everyone just to cover an edge case. Would rather fix it in the contract. Are we sure it actually throws for marketplace v3 too?

Reply:
Yes it's the same issue for v3 as well. Yeah, I agree, fixing it in the contract would be more efficient.